devices/tuya.js

```diff
@@ -432,6 +432,41 @@ module.exports = [
             '3_single', '3_double', '3_hold', '4_single', '4_double', '4_hold'])],
         toZigbee: [],
     },
+    {
+        zigbeeModel: ['TS004F'],
+        model: 'TS004F',
+        vendor: 'TuYa',
+        description: 'Wireless switch with 4 buttons',
+        whiteLabel: [{vendor: '_TZ3000_xabckq1v', model: 'TS004F'}],
+        exposes: [
+            e.battery(),
+            e.action(['1_single', '2_single', '3_single', '3_hold', '4_single', '4_hold'])
+        ],
+        fingerprint: [
+            {modelID: 'TS004F', manufacturerName: '_TZ3000_xabckq1v'},
+        ],
+        fromZigbee: [fz.battery, fz.tuya_on_off_action, fz.tuya_levelctrl],
```

Comment:
Instead of `fz.tuya_on_off_action, fz.tuya_levelctrl` try using `fz.command_on, fz.command_off, fz.command_step, fz.command_move, fz.command_stop`.

Reply:
Changed it and removed the unneeded converters.

Checks fail :(
But it works nicely on my machine and it needs the configureKey: 1 

```
npm test
  shell: /usr/bin/bash -e {0}

> zigbee-herdsman-converters@14.0.160 test /home/runner/work/zigbee-herdsman-converters/zigbee-herdsman-converters
> ZHC_TEST=true jest test

FAIL test/index.test.js (7.254 s)
  ● index.js › Verify definitions

    Contains 'configureKey' while it should only contains: 'disableActionGroup,multiEndpoint,applyRedFix,disableDefaultResponse,enhancedHue,timeout,supportsHueAndSaturation,battery,coverInverted,turnsOffAtBrightness1,pinCodeCount,tuyaThermostatSystemMode,tuyaThermostatPreset,tuyaThermostatPresetToSystemMode,thermostat'

       9 |     for (const elem of array2) {
      10 |         if (!array1.includes(elem)) {
    > 11 |             throw new Error(`Contains '${elem}' while it should only contains: '${array1}'`)
         |                   ^
      12 |         }
      13 |     }
      14 | 

      at containsOnly (test/index.test.js:11:19)
      at forEach (test/index.test.js:269:17)
          at Array.forEach (<anonymous>)
      at Object.<anonymous> (test/index.test.js:179:27)

PASS test/color.test.js
PASS test/otaCommon.test.js
PASS test/utils.test.js

Test Suites: 1 failed, 3 passed, 4 total
Tests:       1 failed, 108 passed, 109 total
Snapshots:   0 total
Time:        8.463 s
Ran all test suites matching /test/i.
npm ERR! Test failed.  See above for more details.
Error: Process completed with exit code 1.
```

Reply:
This is because you are on 1.18.3 and not on 1.18.3-dev (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html). The configureKey isn't needed anymore.

Reply:
Ah, so if i leave an empty 'meta' then it's ok